src/Code/utils/parseCode.ts

```diff
@@ -0,0 +1,42 @@
+// @ts-ignore
+import * as refractor from 'refractor';
+import { lineNumberify } from './lineNumberify';
+
+function parsePlainText(code: string) {
+  return code.split('\n').map((value, i, arr) => ({
+    type: 'element',
+    tagName: 'span',
+    properties: {},
+    children: [
+      {
+        type: 'text',
+        value: arr.length - 1 === i ? value : `${value}\n`,
+      },
+    ],
+  }));
```

Comment:
Note: apparently this returned object almost resembles a React element — can't return just that directly?

Reply:
No, not really... unless we want to introduce plenty of logic.
As you can see, that data goes through `lineNumberify` so if we wanted to have line numbers, we'd need to contain the logic here.
Then, once we have lines logic in `parsePlainText` function, we'd still need to handle that in `parseCode`, and then in actual viewer component (see line 29).
In short - yes we could, but not sure if it's worth it